src/Manager.js

```diff
@@ -612,54 +670,82 @@ class GiveawaysManager extends EventEmitter {
 
     /**
      * @ignore
-     * @param {any} packet
      */
-    async _handleRawPacket(packet) {
-        if (!['MESSAGE_REACTION_ADD', 'MESSAGE_REACTION_REMOVE'].includes(packet.t)) return;
-        if (packet.d.user_id === this.client.user.id) return;
-
-        const giveaway = this.giveaways.find((g) => g.messageId === packet.d.message_id);
-        if (!giveaway || (giveaway.ended && packet.t === 'MESSAGE_REACTION_REMOVE')) return;
+    async _handleEvents() {
+        const checkForDropEnd = async (giveaway) => {
+            const users = await giveaway.fetchAllEntrants().catch(() => {});
+
+            let validUsers = 0;
+            for (const user of [...(users?.values() || [])]) {
```

Comment:
```suggestion
            for (const user of (users?.values() || [])) {
```

`users.values` is already iterable. Constructing array of values here is not necessary.